src/tests/ftest/util/exception_utils.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import os
+from env_modules import show_avail, get_module_list
+from general_utils import run_command
+
+class CommandFailure(Exception):
+    """Base exception for this module."""
+
+class MPILoadError(Exception):
+    """Exception raised when loading an MPI module fails."""
+
+    def __init__(self, module):
+        """Create an MPILoadError object.
+
+        Args:
+            module (str): the name of the module
+            message (str): error explanation
+        """
+        message = "Failed to load an {0} module from the list {3}.\n" \
+                  "Available modules:\n{1}\n" \
+                  "Installed *{0}* RPMs:\n{2}\nEnvironment:\n{4}".format(module, show_avail(),
+                  "\n".join(list(filter(lambda x: "openmpi" in x,
+                                        run_command("rpm -qa").stdout_text.split("\n")))),
+                                        ' '.join(get_module_list(module)),
+                                        "\n".join([f"{k}: {v}" for k, v in
```

Comment:
f-string syntax. E.g. `f"{var}"` was introduced in python 3.6. Do all distros we support (that we need to run our functional test framework on) have python 3.6 available?

Reply:
I don't know, TBH.

Reply:
The only f-string usage I found is in this file: https://github.com/daos-stack/daos/blob/249b09ec00993d49f019407fdc756c966a111704/src/tests/ftest/fault_domain/fault_domain.py#L82-L84
But that test has been skipped for months, so it's not clear whether it works on the VM distros we test (which could implicitly tell us 3.6 is available).

About a year ago, I feel like I was told python 3.6 was too high of a baseline. I would love if we could move it up a bit. There are tons of "modern" features we're unable to use.

Reply:
FWIW, my centos 7 boro node has python 3.6.8

Reply:
CentOS 7 is going to be the long pole, so it seems to meet the minimum.